spanner/session.go

```diff
@@ -813,13 +882,53 @@ func (p *sessionPool) growPoolLocked(numSessions uint64, distributeOverChannels
 	return p.sc.batchCreateSessions(int32(numSessions), distributeOverChannels, p)
 }
 
+func (p *sessionPool) createMultiplexedSession(ctx context.Context) error {
+	for c := range p.multiplexedSessionsReq {
```

Comment:
It took some time for me to understand what this does, but I think I now understand it:
1. During session pool initialization, we call `go createMultiplexedSession(..)`.
2. That goroutine is then blocked on this `for` statement until there is a request for a multiplexed session.
3. The request can come from a read-only transaction (if multiplexed sessions are enabled), or from the background worker keeping the multiplexed session fresh.

That means that:
1. The multiplexed session creation is __always__ on the critical path of the first read-only transaction.
2. The goroutine is stuck here forever if there is never a request for a multiplexed session.

I don't think that is what we would want. Instead, we should try to:
1. Always start the creation of a multiplexed session as soon as possible if multiplexed sessions are enabled. This means that the creation will not be on the critical path of the first read-only transaction.
2. Not call this method at all if multiplexed sessions are disabled.

I think that many of my above comments are also a direct consequence of this behavior.

Reply:
> Always start the creation of a multiplexed session as soon as possible if multiplexed sessions are enabled. This means that the creation will not be on the critical path of the first read-only transaction.


I updated to trigger createMultiplexedSession create session pool



> Not call this method at all if multiplexed sessions are disabled.

Updated, now we spawn this background thread only when mux is enabled